Lib/asyncio/tasks.py

```diff
@@ -439,7 +439,7 @@ async def wait_for(fut, timeout):
                 # after wait_for() returns.
                 # See https://bugs.python.org/issue32751
                 await _cancel_and_wait(fut, loop=loop)
-                raise
+                return fut.result()
```

Comment:
This ensures that a cancellation suppressed in the inner future, also suppresses it in `wait_for()` as one would expect.